components/reusable-components/footer/Footer.tsx

```diff
@@ -1,69 +1,173 @@
-import React from 'react';
-import Link from 'next/link';
-import Image from 'next/image';
-import 'bootstrap-icons/font/bootstrap-icons.css';
+'use client';
 
+import React, { useState } from 'react';
+import * as Yup from 'yup';
+import { useFormik } from 'formik';
+import { toast } from 'react-toastify';
+import 'react-toastify/dist/ReactToastify.css';
+import Image from 'next/image';
+import Link from 'next/link';
+import { useTheme } from '../../../app/context/themeContext';
 import styles from './footer.module.scss';
-import EmailForm from '../newsletter/emailform/emailform';
+import TextInput from '../text-input/TextInput';
+import discord from '../../../public/icons-footer/discord.svg';
+import linkedin from '../../../public/icons-footer/linkedIn.svg';
+import github from '../../../public/icons-footer/github.svg';
+import facebook from '../../../public/icons-footer/facebook.svg';
+import instagram from '../../../public/icons-footer/instagram.svg';
+import youtube from '../../../public/icons-footer/youtube.svg';
+import logolight from '../../../public/logo/logo-white.svg';
+import logodark from '../../../public/logo/logo-black.svg';
+
+const Footer: React.FC = () => {
+  const { theme } = useTheme();
+  const initialValues = { name: '', email: '' };
+  const validationSchema = Yup.object({
+    name: Yup.string()
+      .matches(/^[a-zA-Z .'-]+$/, 'Невалидно име!')
+      .min(2, 'Вашето име е премногу кратко!')
+      .max(50, 'Вашето име е премногу долго!')
+      .required('Задолжително внесете име'),
+    email: Yup.string()
+      // eslint-disable-next-line no-unused-vars
+      .email('Невалидна емаил адреса')
+      .required('Задолжително внесете емаил адреса'),
+  });
+  // eslint-disable-next-line no-unused-vars
+  const [successMessage, setSuccessMessage] = useState<boolean>(false);
+  // eslint-disable-next-line no-unused-vars
+  const [errorMessage, setErrorMessage] = useState<boolean>(false);
+
+  // eslint-disable-next-line no-unused-vars
+  const handleSubmit = async (values: { name: string; email: string }) => {
+    try {
+      // Simulating API call
+      /*
+      const response = await fetch('your-backend-api-endpoint', {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify(values),
+      });
+      
+      if (response.ok) {
+        setSuccessMessage(true);
+      } else {
+        setErrorMessage(true);
+      }
+      */
+      // Simulating success
```

Comment:
you can call our API here.
check the submitContactForm for example of how you can do this
the API Documentation:  **https://staging-api.learnhub.mk/docs/**
Since this is still not implemented by the backend team it is good to just extract the function in a separate file and we will test once the backend team is done